### DISCUSSION TITLE
Rework Shortcut Keys to Industry standards

### DISCUSSION THREAD

**Discussion by philippedcote** [godotengine/godot-proposals#4098] [Editor]
There are some things in Godot that don’t stick to industry standards (but it should). Here is a very short list:

- `Command + A` is used to add an object rather than "Select All"
- `Command + N` doesn’t do anything, which is puzzling.
- `Command + Z` very often doesn’t do anything (looks like a bug though)
- Many Commands don’t have a proper menu above

I could go on and on but you see the point.

**Comment by Calinou** [comment]
> Command + A is used to add an object rather than "Select All"

Selecting all nodes in the scene tree dock isn't needed often, but adding new nodes quickly is done much more often. Changing <kbd>Ctrl + A</kbd> to something else would break existing users' muscle memory – is it worth it?

> Command + N doesn’t do anything, which is puzzling.

A shortcut was added for New Scene in 4.0 and 3.5beta, see https://github.com/godotengine/godot/pull/48574 and its `3.x` backport https://github.com/godotengine/godot/pull/57011.

> Command + Z very often doesn’t do anything (looks like a bug though)

This is a known bug: https://github.com/godotengine/godot/issues/53424

**Reply by philippedcote** [reply]
For adding Nodes, why not use `Shift + A` like Blender instead?

**Reply by Calinou** [reply]
Shortcuts that only use <kbd>Shift</kbd> as a modifier always felt uncanny to me. It's uncommon for software to use those in general, especially as they conflict with text fields (since <kbd>Shift + A</kbd> inserts an uppercase A character there).

**Reply by philippedcote** [reply]
Ha, you are right! :) Blender is one of those softwares that was, for a long time, far away from industry standards. I guess this shortcut is a reminder of that.

Here’s a tentative proposal:

- `Command + N` = New Node
- `Command + Shift + N` = New Scene

The reflexion behind is that you need Nodes more often than Scenes. Also, both are related, hierarchy-speaking.

**Reply by AaronRecord** [reply]
> Here’s a tentative proposal:
> 
> * `Command + N` = New Node
> * `Command + Shift + N` = New Scene
> 
> The reflexion behind is that you need Nodes more often than Scenes. Also, both are related, hierarchy-speaking.

<kbd>Command + A</kbd> is still easier to reach though, as it can be pressed with just the left hand with the right hand still on the mouse.

**Comment by setanarut** [comment]
These are the standard shortcuts that everyone is used to. Godot macOs default key suggestions;

| Key                       | Action                                        |
|---------------------------|-----------------------------------------------|
| `Command+w`                 | Close window, tab, or panel                      |
| `Return`                    | Rename file or Node (**no double click! no F2!**) |
| `Command+o` or `Double Click` | Open Node (show in editor viewport)           |

### Also
This is better than `Command+Backspace` and `Shift+Delete` because Macbooks don't have the `delete` key. `Fn+Shift+Backspace` is impossible to press with one hand.;
<img width="551" alt="Ekran Resmi 2022-02-27 19 45 07" src="https://user-images.githubusercontent.com/36481442/155891481-13f7d8b0-4cbc-4254-81db-961e971c57f7.png">

edit:
> Rename with return was applied in Godot 4.
